engine/apps/api/serializers/shift_swap.py

```diff
@@ -0,0 +1,73 @@
+import datetime
+
+from django.utils import timezone
+from rest_framework import serializers
+
+from apps.schedules.models import OnCallSchedule
+from apps.shift_swaps.models import ShiftSwapRequest
+from apps.user_management.models import User
+from common.api_helpers.custom_fields import OrganizationFilteredPrimaryKeyRelatedField, TimeZoneAwareDatetimeField
+from common.api_helpers.utils import CurrentUserDefault
+
+
+class ShiftSwapRequestSerializer(serializers.ModelSerializer):
+    id = serializers.CharField(read_only=True, source="public_primary_key")
+    schedule = OrganizationFilteredPrimaryKeyRelatedField(queryset=OnCallSchedule.objects)
+
+    created_at = TimeZoneAwareDatetimeField(read_only=True)
+    updated_at = TimeZoneAwareDatetimeField(read_only=True)
+
+    swap_start = TimeZoneAwareDatetimeField()
+    swap_end = TimeZoneAwareDatetimeField()
+
+    beneficiary = OrganizationFilteredPrimaryKeyRelatedField(queryset=User.objects, default=CurrentUserDefault())
+    benefactor = OrganizationFilteredPrimaryKeyRelatedField(queryset=User.objects, required=False, allow_null=True)
+
+    class Meta:
+        model = ShiftSwapRequest
+        fields = [
+            "id",
+            "created_at",
+            "updated_at",
+            "status",
+            "schedule",
+            "swap_start",
+            "swap_end",
+            "description",
+            "beneficiary",
+            "benefactor",
+        ]
+        read_only_fields = [
+            "status",
```

Comment:
are these fields readonly too?
```suggestion
            "id",
            "created_at",
            "updated_at",
            "status",
```

Reply:
these three fields are marked `read_only=True`, inside of their fields above. I _think_ that should infer the same meaning, no?